src/fittable_models.jl

```diff
@@ -0,0 +1,45 @@
+# Convenience function for wavelength conversion
+aa_to_invum(wave::Real) = 10000 / wave
+aa_to_invum(wave::Quantity) = aa_to_invum(ustrip(u"angstrom", wave))
+
+"""
+    FM90(C1 = 0.10, C2 = 0.70, C3 = 3.23, C4 = 0.41, xo = 4.60, γ = 0.9)(λ::Real)
+
+The parameters C1, C2, C3, C4, xo and γ are y-intercept, slope, bump amplitude, FUV rise amplitude,
+bump centroid and bump width respectively.
+
+Fitzpatrick & Massa (1990) 6 parameter ultraviolet shape model, this model is only applicable at UV wavelengths.
+
+If `λ` is a `Unitful.Quantity` it will be automatically converted to Å and the
+returned value will be `UnitfulAstro.mag`.
+"""
+struct FM90
+    C1::Real
+    C2::Real
+    C3::Real
+    C4::Real
+    xo::Real
+    γ::Real
+end
+
+FM90() = FM90(0.10, 0.70, 3.23, 0.41, 4.60, 0.9)
+
+function (z::FM90)(λ::Real)
+    x = aa_to_invum(λ)
+
+    if !(1.0 / 0.32 <= x <= 1.0 / 0.0912)
+        return zero(eltype(x))
```

Comment:
This can cause type instability, because x may be an integer but everything we normally return is a float. For now `0.0x` has worked well.

Reply:
As an alternative
```suggestion
        return zero(float(eltype(x)))
```
could give the zero of the corresponding float type

Reply:
Okay, I will fix it. Do you have any good resource to learn about type stability and how to fix instabilities? @giordano @mileslucas

Reply:
Mostly just practice. There’s a few tools for testing stuff- @code_warntype, @inferred, Cthulu.jl

I think the harder question is knowing when you absolutely need to apply and and when it’s not worth the trouble.